File: examples/go/minimal-order-service/adapters/outbound/fake_payment_gateway.go
package outbound

import (
	"github.com/google/uuid"
	"github.com/example/order-service/domain"
)

// FakePaymentGateway is a fake implementation of PaymentGateway for testing.
type FakePaymentGateway struct{}

// NewFakePaymentGateway creates a new FakePaymentGateway.
func NewFakePaymentGateway() *FakePaymentGateway {
	return &FakePaymentGateway{}
}

// ProcessPayment processes a payment and always succeeds.
func (g *FakePaymentGateway) ProcessPayment(amount *domain.Money, orderID string) (*PaymentResult, error) {
	transactionID := "txn-" + uuid.New().String()
	return &PaymentResult{
		Success:       true,
		TransactionID: transactionID,
	}, nil
}

File: examples/cpp/minimal-order-service/adapters/outbound/fake_payment_gateway.cpp
#include "adapters/outbound/fake_payment_gateway.hpp"
#include <random>
#include <sstream>
#include <iomanip>

namespace adapters::outbound {

application::ports::outbound::PaymentResult FakePaymentGateway::processPayment(
    std::shared_ptr<domain::Money> amount,
    const std::string& orderId
) {
    // Generate a fake transaction ID
    std::random_device rd;
    std::mt19937 gen(rd());
    std::uniform_int_distribution<> dis(0, 15);
    
    std::ostringstream oss;
    oss << "txn-";
    for (int i = 0; i < 32; ++i) {
        oss << std::hex << dis(gen);
    }
    
    return application::ports::outbound::PaymentResult::successResult(oss.str());
}

} // namespace adapters::outbound
